src/preact/utils/url.js

```diff
@@ -1,5 +1,15 @@
 import {INVALID_PROTOCOLS} from '#core/types/string/url';
 
+/**
+ * @type {{
+ *   anchor_?: HTMLAnchorElement,
+ *   getAnchor_: function(): HTMLAnchorElement
+ *   parse: function(string|URL): URL
+ *   isProtocolValid: function(URL|string): boolean
```

Comment:
consider `/** @typedef {URL|string} UrlLike */` for this file/`#core/types/string/url`